hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TokenUpdatePrecompile.java

```diff
@@ -113,9 +125,65 @@ public void run(MessageFrame frame) {
         validateTrue(updateOp.tokenID() != null, INVALID_TOKEN_ID);
         tokenId = Id.fromGrpcToken(updateOp.tokenID());
         type = UPDATE_TOKEN_INFO;
+
+        if (updateOp.treasury() != null) {
+            final var treasuryId = Id.fromGrpcAccount(updateOp.treasury());
+            final var treasuryHasSigned =
+                    KeyActivationUtils.validateKey(
+                            frame,
+                            treasuryId.asEvmAddress(),
+                            sigsVerifier::hasActiveKey,
+                            ledgers,
+                            aliases);
+
+            validateTrue(
+                    treasuryHasSigned,
+                    INVALID_FULL_PREFIX_SIGNATURE_FOR_PRECOMPILE,
+                    "Token update");
```

Comment:
Should we add a more meaningful message? Like "Treasury account signature missing in token update?"?